core/src/layers/prometheus.rs

```diff
@@ -604,26 +663,39 @@ pub struct PrometheusMetricWrapper<R> {
     op: Operation,
     stats: Arc<PrometheusMetrics>,
     scheme: String,
+    path: String,
 }
 
 impl<R> PrometheusMetricWrapper<R> {
-    fn new(inner: R, op: Operation, stats: Arc<PrometheusMetrics>, scheme: &String) -> Self {
+    fn new(
+        inner: R,
+        op: Operation,
+        stats: Arc<PrometheusMetrics>,
+        scheme: &String,
```

Comment:
Use `Scheme` whenever possible.

Reply:
I think scheme: &String here would be a better choice

1. keep same style with PrometheusAccessor
2. we only need scheme to be a string in this layer, this means we always need a value to store scheme as string to avoid `scheme.to_string()` every time

Reply:
Scheme is an enum that cheap to clone, no extra allocation needed. So it's better to store Scheme everywhere snd convert to `&'static str` while needed.